### DISCUSSION TITLE
Facing issue when redirecting back on the app route from the different server

### DISCUSSION THREAD

**Discussion by asada-chetu** [Kyon147/laravel-shopify#99] [Q&A]
Facing issues when redirecting back on the app route from a different server. we are opening a third-party page in the embedded app iframe after completing the process. we want to redirect back to the specific route with the host but show the blank page in the embedded app iframe. Please guide me what's the issue.

   
![image](https://user-images.githubusercontent.com/102379477/220431891-8e621e3d-9af0-43ac-b82f-c2e895326689.png)

**Comment by asada-chetu** [comment]
@Kyon147 Please provide us solution regarding the above redirection issue from a third-party domain.

**Comment by Kyon147** [comment]
Hi @asada-chetu 

This is quite a unique use case, you would also need to check with Shopify if this flow is allowed. As they don't allow anything to happen after OAuth apart from you going to your home page of the app.

Why are you going to a third party page on another server? Can you not stay within the app and just make an api call that you can do from within it to do the same logic?

**Comment by dandyleon19** [comment]
Hello @Kyon147 ,  I have the same problem, within the app I am going to a third-party app to authenticate a service, but when I return to the shopify app, it seems that the session is lost.

When I want to return to the home page, it just keeps loading, it seems that the host would need to be passed to it in the redirection, but I can't find a way to get that host

![image](https://user-images.githubusercontent.com/48505581/232165387-67c9f920-7cc8-47db-8f2e-5300859d6bf7.png)

![image](https://user-images.githubusercontent.com/48505581/232164971-cfd72c13-8623-4a63-aa85-33811bbbac56.png)

**Comment by dandyleon19** [comment]
Hi @asada-chetu , can you help me how you retrieve the host please?

In your case i think should be redirect to the "home" page this way, adding the "shop" and "host": 

![image](https://user-images.githubusercontent.com/48505581/232166676-f5732d53-5c07-4cbb-af60-c7ca54a67213.png)

I would ask you please help me how to get the host, thanks!!
